api_server/src/http_service.rs

```diff
@@ -674,6 +674,9 @@ fn describe(method: &Method, path: &str, body: &Option<String>) -> String {
 }
 
 #[cfg(test)]
+// Allowing assertions on constants so we can check enum variants using
+// match instead of equals.
+#[allow(clippy::assertions_on_constants)]
```

Comment:
Just wondering, is this required only for things like `assert!(false)`, or there are more complex examples of assertions which aren't allowed otherwise?

Reply:
Only assert true or assert false.